mlflow/doctor.py

```diff
@@ -0,0 +1,39 @@
+import os
+import json
+import pkg_resources
+import platform
+import click
+import mlflow
+
+
+def doctor():
```

Comment:
Output:

```
Platform: Linux-5.15.0-52-generic-x86_64-with-debian-bullseye-sid
Python version: 3.7.10
MLflow version: 2.0.0.dev0
MLflow module location: /home/haru/Desktop/repositories/mlflow/mlflow/__init__.py
Tracking URI: file:///home/haru/Desktop/repositories/mlflow/mlruns
Registry URI: file:///home/haru/Desktop/repositories/mlflow/mlruns
Active experiment ID: 0
Active run ID: b55add8ee08d4a489901c725bfccbd53
Active run artifact URI: file:///home/haru/Desktop/repositories/mlflow/mlruns/0/b55add8ee08d4a489901c725bfccbd53/artifacts
MLflow environment variables: {
  "MLFLOW_FOO": "yes"
}
MLflow dependencies: {
  "click": "8.0.1",
  "cloudpickle": "1.6.0",
  "databricks-cli": "0.14.3",
  "entrypoints": "0.3",
  "gitpython": "3.1.18",
  "pyyaml": "6.0",
  "protobuf": "3.19.0",
  "pytz": "2022.5",
  "requests": "2.28.1",
  "packaging": "21.3",
  "importlib_metadata": "5.0.0",
  "sqlparse": "0.4.3",
  "alembic": "1.8.1",
  "docker": "4.4.4",
  "Flask": "2.2.2",
  "numpy": "1.21.3",
  "scipy": "1.7.1",
  "pandas": "1.3.5",
  "prometheus-flask-exporter": "0.20.3",
  "querystring_parser": "1.2.4",
  "sqlalchemy": "1.4.42",
  "gunicorn": "20.1.0"
}
> python -c "import mlflow; mlflow.start_run(); mlflow.doctor()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/haru/Desktop/repositories/mlflow/mlflow/doctor.py", line 18, in doctor
    ("Python version", platform().python_version),
TypeError: 'module' object is not callable
> python -c "import mlflow; mlflow.start_run(); mlflow.doctor()"
Platform: Linux-5.15.0-52-generic-x86_64-with-debian-bullseye-sid
Python version: 3.7.10
MLflow version: 2.0.0.dev0
MLflow module location: /home/haru/Desktop/repositories/mlflow/mlflow/__init__.py
Tracking URI: file:///home/haru/Desktop/repositories/mlflow/mlruns
Registry URI: file:///home/haru/Desktop/repositories/mlflow/mlruns
Active experiment ID: 0
Active run ID: 896b2e1d5bda4a71aba1c59bc4e0056e
Active run artifact URI: file:///home/haru/Desktop/repositories/mlflow/mlruns/0/896b2e1d5bda4a71aba1c59bc4e0056e/artifacts
MLflow environment variables: {
  "MLFLOW_FOO": "yes"
}
MLflow dependencies: {
  "click": "8.0.1",
  "cloudpickle": "1.6.0",
  "databricks-cli": "0.14.3",
  "entrypoints": "0.3",
  "gitpython": "3.1.18",
  "pyyaml": "6.0",
  "protobuf": "3.19.0",
  "pytz": "2022.5",
  "requests": "2.28.1",
  "packaging": "21.3",
  "importlib_metadata": "5.0.0",
  "sqlparse": "0.4.3",
  "alembic": "1.8.1",
  "docker": "4.4.4",
  "Flask": "2.2.2",
  "numpy": "1.21.3",
  "scipy": "1.7.1",
  "pandas": "1.3.5",
  "prometheus-flask-exporter": "0.20.3",
  "querystring_parser": "1.2.4",
  "sqlalchemy": "1.4.42",
  "gunicorn": "20.1.0"
}
```

Reply:
Should we auto-obfuscate environment variables that users typically obscure when posting information and instead print the `type()`?
`MLFLOW_S3_ENDPOINT_URL` / `MLFLOW_S3_UPLOAD_EXTRA_ARGS` / `MLFLOW_KERBEROS_USER` and perhaps any other values that might define the physical layout of a user's infra or identifying characteristics of their environment?

Reply:
How do we select environment variables to auto-obfuscate? Manually?

Reply:
We could provide an optional flag argument to hide environment variables.

Reply:
Do we need to report types? They are all strings, aren't they?

Reply:
`MLFLOW_REQUIREMENTS_INFERENCE_TIMEOUT`, `MLFLOW_SQLALCHEMYSTORE_MAX_OVERFLOW`, `MLFLOW_SQLALCHEMYSTORE_POOL_RECYCLE`, `MLFLOW_SQLALCHEMYSTORE_POOL_SIZE`, MLFLOW_GCS_DEFAULT_TIMEOUT`, `MLFLOW_GCS_UPLOAD_CHUNK_SIZE`, `MLFLOW_GCS_DOWNLOAD_CHUNK_SIZE`, `MLFLOW_HTTP_REQUEST_TIMEOUT`, `MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR`, `MLFLOW_HTTP_REQUEST_MAX_RETRIES` are all int

`MLFLOW_TRACKING_AWS_SIGV4`, `MLFLOW_SQLALCHEMYSTORE_ECHO`, `MLFLOW_S3_IGNORE_TLS`, `MLFLOW_DISABLE_ENV_MANAGER_CONDA_WARNING` are bool

and the rest are str

Reply:
btw what is the purpose of printing types?

Reply:
Just so that we can see that something has been overridden in an env arg (or should be, but is not).